lib/train/platforms/detect/helpers/os_common.rb

```diff
@@ -52,5 +52,11 @@ def unix_uname_m
       return @uname[:m] if @uname.key?(:m)
       @uname[:m] = command_output('uname -m')
     end
+
+    def cisco_show_version
+      res = command_output('show version')
```

Comment:
Can we memoify this so we can call `cisco_show_version` multiple times in a run without running the command multiple times?

Perhaps something like this:

```ruby
return @cisco_show_version if defined?(@cisco_show_version)

res = command_output('show version')
m = res.match(/^Cisco IOS Software, [^,]+? \(([^,]+?)\), Version (\d+\.\d+)/)
@cisco_show_version = m.nil? ? nil : { version: m[2], model: m[1], raw: res }
```

Reply:
i got a follow-up that does that in a more general way; if we can get this merged I'll follow i tup